images/echoserver/echoserver.go

```diff
@@ -121,3 +161,63 @@ func processError(w http.ResponseWriter, err error, code int) {
 	w.WriteHeader(code)
 	w.Write(body)
 }
+
+func listenAndServeTLS(addr string, serverCert string, serverPrivKey string, clientCA string, handler http.Handler) error {
+	var config tls.Config
+
+	// optionally enable client certificate validation if client CA certificates are given
+	if clientCA != "" {
+		ca, err := ioutil.ReadFile(clientCA)
+		if err != nil {
+			return err
+		}
+
+		certPool := x509.NewCertPool()
+		if ok := certPool.AppendCertsFromPEM(ca); !ok {
+			return fmt.Errorf("unable to append certificate in %s to CA pool", clientCA)
```

Comment:
Prefer `%q` for errors (it reduces potential ambiguity).

Reply:
Fixed.